Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Services/ApiGatewayRouteConfigService.cs

```diff
@@ -0,0 +1,430 @@
+    ///   </item>
+    ///   <item>
+    ///     <term>MatchedSegmentsBeforeGreedy</term>
+    ///     <description>The number of segments (literal or normal variable) that were matched before encountering any greedy variable segment. A higher number indicates a more specific match before resorting to greedily matching the remainder of the path.</description>
+    ///   </item>
+    /// </list>
+    /// </returns>
+    private (
+        bool Matched, 
+        int LiteralMatches, 
+        int VariableCount, 
+        int GreedyCount, 
+        int MatchedSegmentsBeforeGreedy)
+        MatchRoute(string[] routeSegments, string[] requestSegments)
+    {
+        var routeTemplateIndex = 0;
+        var requestPathIndex = 0;
+        var literalMatches = 0;
+        var variableCount = 0;
+        var greedyCount = 0;
+        var matched = true;
+
+        var matchedSegmentsBeforeGreedy = 0;
+        var encounteredGreedy = false;
+
+        while (
+            matched && 
+            routeTemplateIndex < routeSegments.Length && 
+            requestPathIndex < requestSegments.Length)
+        {
+            var routeTemplateSegment = routeSegments[routeTemplateIndex];
+            var requestPathSegment = requestSegments[requestPathIndex];
+
+            if (IsVariableSegment(routeTemplateSegment))
+            {
+                if (IsGreedyVariable(routeTemplateSegment))
+                {
+                    // Greedy variable must match at least one segment
+                    // Check if we have at least one segment remaining
+                    if (requestPathIndex >= requestSegments.Length)
+                    {
+                        // No segments left to match the greedy variable
+                        matched = false;
+                    }
+                    else
+                    {
+                        greedyCount++;
+                        encounteredGreedy = true;
+                        routeTemplateIndex++;
+                        // Greedy matches all remaining segments
+                        requestPathIndex = requestSegments.Length; 
+                    }
+                }
+                else
+                {
+                    variableCount++;
+                    if (!encounteredGreedy) matchedSegmentsBeforeGreedy++;
+                    routeTemplateIndex++;
+                    requestPathIndex++;
+                }
+            }
+            else
+            {
+                if (!routeTemplateSegment.Equals(requestPathSegment, StringComparison.OrdinalIgnoreCase))
+                {
+                    matched = false;
+                }
+                else
+                {
+                    literalMatches++;
+                    if (!encounteredGreedy) matchedSegmentsBeforeGreedy++;
+                    routeTemplateIndex++;
+                    requestPathIndex++;
+                }
+            }
+        }
+
+        // If there are leftover route segments
```

Comment:
Can you explain what would be an example route and request path that would go into this section. It would be good to add examples in the comment block to help readers. I'm struggling to read this and understand in what scenario would the route ever be a matched if there are more route segments then were in the request path.

Reply:
I simplified this code paths as it was redundant. This was a leftover from my tweaking when I was working on it. I also added comments and examples explaining the code paths as much as possible.